tests/test_metadata.py

```diff
@@ -0,0 +1,72 @@
+import pytest
+
+
+class TestMetadata(object):
+
+    def test_invalid_name(self, w_checkout):
+        # TODO: It's a value error in dataset and branch
```

Comment:
@rlizzo Why do you think it's a SyntaxError? Illegal names in dataset creation and branch creation raises ValueError

Reply:
You're right this should be a value error.